chain/chunks/src/lib.rs

```diff
@@ -272,8 +273,9 @@ impl SealsManager {
     }
 
     fn approve_chunk(&mut self, chunk_hash: &ChunkHash) {
-        let seal = self.active_demurs.remove(chunk_hash).expect("seal should be already produced");
-        Self::insert_past_seal(&mut self.past_seals, seal.height, chunk_hash.clone());
+        if let Some(seal) = self.active_demurs.remove(chunk_hash) {
```

Comment:
Of course! It is no longer true that the active demur will necessarily be there when the chunk is reconstructed because it may have expired and moved into the past seals. I should have removed the `expect` in the original PR.

I think it is good to check that there is an entry in `past_seals` though, since this would be the case if the demur expired recently. Please see https://github.com/nearprotocol/nearcore/pull/2788